app/src/main/java/lt/vilnius/tvarkau/fragments/ReportImportDialogFragment.java

```diff
@@ -207,6 +208,81 @@ private boolean validateVilniusSignInputFields() {
         return emailIsValid && passwordIsValid;
     }
 
+    private void loadUserReportsFromVilniusAccount(String email, Dialog dialog) {
+
+        GetProblemsParams params = new GetProblemsParams.Builder()
+            .setStart(0)
+            .setLimit(100)
+            .setDescriptionFilter(null)
+            .setTypeFilter(null)
+            .setAddressFilter(null)
+            .setReporterFilter(email)
+            .setDateFilter(null)
+            .setStatusFilter(null)
+            .create();
+
+        ApiRequest<GetProblemsParams> request = new ApiRequest<>(ApiMethod.GET_PROBLEMS, params);
+
+        Action1<ApiResponse<List<Problem>>> onSuccess = apiResponse -> {
+            if (apiResponse.getResult().size() > 0) {
+                List<Problem> vilniusAccountReports = new ArrayList<>();
+                vilniusAccountReports.addAll(apiResponse.getResult());
+                for (Problem report : vilniusAccountReports) {
+                    String reportId = report.getIdForVilniusAccount();
+                    if (!myProblemsPreferences.getAll().isEmpty()) {
+                        for (String key : myProblemsPreferences.getAll().keySet()) {
+                            if (!reportId.equals(myProblemsPreferences.getString(key, ""))) {
+                                myProblemsPreferences
+                                    .edit()
+                                    .putString(NewProblemActivity.PROBLEM_PREFERENCE_KEY + reportId, reportId)
+                                    .apply();
+                            }
+                        }
+                    } else {
+                        myProblemsPreferences
+                            .edit()
+                            .putString(NewProblemActivity.PROBLEM_PREFERENCE_KEY + reportId, reportId)
+                            .apply();
+                    }
+                }
+                VilniusSignInListener listener = (VilniusSignInListener) getActivity();
```

Comment:
Next line is pretty short, so I would not define new local variable `listener`. You can write like this, and it's still pretty good readable:
`((VilniusSignInLisetner) getActivity()).onVilniusSingIn();`